### ISSUE TITLE
when speaker changes the music does not change to the new speaker to play

### ISSUE THREAD

**Issue by tommady** [hrkfdn/ncspot#339]
Hi, sorry first if this question is not proper to ask here 🙇🏻 

just like the title described, when I change the speaker I need to quit the ncspot and reopen it again.

I am using macOS and I did not test on Linux
ncspot version is 0.2.4 installed from homebrew

steps:
1. open the ncspot with a speaker and playing a song
![image](https://user-images.githubusercontent.com/11532828/101017322-e488a080-35a4-11eb-985e-245ca6924e4d.png)

2. change the speaker to another one
![image](https://user-images.githubusercontent.com/11532828/101017407-fff3ab80-35a4-11eb-815c-8b4934aef277.png)

the music still coming out from the first speaker (the VX3211-4K).

but there has an exception when I use the airpod pro connected with the Bluetooth connected then open the ncspot to play music, then I take off the airpod while the music still playing and the Bluetooth disconnected, the music will come out from the speaker.

but this exception does not work with the reverse steps.

please and thanks if you could help me to solve this.

**Comment by JojiiOfficial** [comment]
Can't reproduce it since I don't have macOS but it is working properly on Linux.

**Comment by hgvhgv** [comment]
I have what seems to be the same issue: could be related to https://github.com/RustAudio/rodio/issues/327? But the version of rodio included with librespot is .9, so behind this report.

I'm on macos 10.15.7 and installed ncspot via brew. Rodio shows up as the backend in debug but nothing gets listed there when I change the audio source in macos.

I will try building from source with other backends when I have a chance. There is no way to install from brew with other backends, right?

**Comment by hrkfdn** [comment]
I'm not sure if this can be handled in ncspot. Could you check if you can reproduce this issue with librespot?

**Comment by hgvhgv** [comment]
If it's a problem with rodio then it would make sense that it's an issue with librespot and ncspot can't/shouldn't handle it. I just wanted to note that I had the same issue in case others came to look.

I might not have time for a while to build librespot (or ncspot with a different backend) to test this out (I'm just an amateur messing around who also hates the spotify interface). And the immediate solution (quit and restart ncspot) works just fine; I just save my queue as a temp playlist.

Thanks for the quick reply and the amazing client! It's really so good!

**Comment by hgvhgv** [comment]
So I recompiled ncspot with the portaudio backend and had the same problem; I then also recompiled librespot with portaudio and confirmed the problem also persisted there (https://github.com/librespot-org/librespot/issues/547). So this is definitely an upstream problem, and I'll close here.

I did, however, realize that the rodio backend has a lot of noise / crackling and that portaudio is much improved, just fyi to those interested (e.g., https://github.com/hrkfdn/ncspot/issues/333; see also the fix in the works at rodio https://github.com/RustAudio/rodio/issues/225).

If I get around to teaching myself how to put together brew formulae, I'll try to make one with portaudio as the default backend for macos, as it seems more stable than rodio (or at least make backend the option).
